mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -15,20 +17,31 @@
                                 :direction nil}
                :cmd-sequence nil})
 
+;; (defn get-input
+;;   "Read one line of input stream. Trim and convert to upper-case."
+;;   []
+;;   (let [input (string/trim (read-line))]
+;;     (string/upper-case input)))
+
 (defn get-input
-  "Read one line of input stream. Trim and convert to upper-case."
-  []
-  (let [input (string/trim (read-line))]
-    (string/upper-case input)))
+  "Prompt (if provided) and read one line of input stream. Trim and convert to upper-case."
+  ([] (get-input ""))
+  ([prompt]
+   (if (not-empty prompt)
+     (println prompt))
+   (let [input (string/trim (read-line))]
+     (string/upper-case input))))
 
 (defn user-input
   "Reads 3 lines of input from user and calls 3 functions which add this data to info-map"
-  []
-  (println "Please enter plateau-size, initial rover position and command sequence on 3 separate lines:")
-  (let [plateau-size (get-input)
-        rover-position (get-input)
-        cmd-sequence (get-input)]
-    (add-cmd-sequence cmd-sequence (add-rover-position rover-position (add-plateau-size plateau-size info-map)))))
+  [] 
+  (let [plateau-size (get-input "Enter plateau size:")
+        rover-position (get-input "Enter initial rover position:")
+        cmd-sequence (get-input "Enter command sequence:")]
+    (->> info-map
+         (add-plateau-size plateau-size)
+         (add-rover-position rover-position)
+         (add-cmd-sequence cmd-sequence))))
```

Comment:
[GIT HYGIENE]
Was going through the PR commit-by-commit. This refactor doesn't really belong in `Add ability to pass user prompts to get-input`. 
Let's pull this change out into a new commit?